### PR TITLE
fix/ADP-1973 - Db-sync circulating supply calculation

### PR DESCRIPTION
**Pull request by lgobbi-atix**
# Context

Circulating supply query is returning bigger values than the total and max supply

https://github.com/input-output-hk/cardano-js-sdk/pull/281#discussion_r902707585

# Proposed Solution
There is an open issue in db-sync regarding the `ada_pots` table https://github.com/input-output-hk/cardano-db-sync/issues/942
- Add a test to validate that `max supply > total supply > circulating supply` with `.skip` because of the db-sync issue
- Simplify and improve performance of `findCirculatingSupply` and `findTotalSupply` queries
- Add tests for `NetworkInfoBuilder`
- Modify dump-db script to populate `tx_in` and `tx` test tables with missing inputs for outputs that were actually spent

### CODE REVIEW

**Review by rhyslbw — CHANGES_REQUESTED** [review]
58bbb0306fa940297c66f7488ebe295d21e8543d is typed as a `fix`, but then the message indicates it could be a refactor. If it is a fix, please describe the resolution. If it's just a performance improvement, please re-type as `refactor`

**Review by rhyslbw — DISMISSED** [review]
Nice work @lgobbi-atix :handshake:

**Review by mkazlauskas — APPROVED** [review]
Looks good!

Note: didn't look at test data commit

File: packages/cardano-services/test/NetworkInfo/NetworkInfoHttpService.test.ts [hunk 1]
```diff
@@ -320,11 +321,19 @@ describe('NetworkInfoHttpService', () => {
...
+        expect(response.circulating).toBeGreaterThan(0n);
+        expect(response.total).toBeLessThan(maxLovelaceSupply);
+      });
+
+      // FIXME: There is a db-sync issue with the values in `ada_pots` table:
+      //        https://github.com/input-output-hk/cardano-db-sync/issues/942
+      it.skip('lovelaceSupply (with better assertions)', async () => {
+        const { maxLovelaceSupply } = await loadGenesisData(cardanoNodeConfigPath);
+        const response = await provider.lovelaceSupply();
+        expect(response.total).toBeGreaterThan(response.circulating);
```
Comment (ssong-iohk) [inline packages/cardano-services/test/NetworkInfo/NetworkInfoHttpService.test.ts:335]:
```suggestion
        expect(response.total).toBeGreaterThan(response.circulating); // This assertion is to be at the above test case. (i.e. Add at line 327)
```

**Review by ssong-iohk — CHANGES_REQUESTED** [review]

File: packages/cardano-services/test/NetworkInfo/NetworkInfoHttpService.test.ts (hunk 1 above)
Comment (rhyslbw) [reply packages/cardano-services/test/NetworkInfo/NetworkInfoHttpService.test.ts:336]: Done in https://github.com/input-output-hk/cardano-js-sdk/pull/337/commits/d4b18688bcab0cb6b80ccf44c8f1cbc22705cca4

**Review by ssong-iohk — DISMISSED** [review]
LGTM

**Review by rhyslbw — APPROVED** [review]

**Pull request merged**